### DISCUSSION TITLE
Practice on history size and when to use continue as new

### DISCUSSION THREAD

**Discussion by wenerme** [uber/cadence#4629] [Q&A] [answered]
A workflow use user message as signal, if every message result in a ContinueAsNew will cause too many workflow, but all message in one workflow will cause very large history, so I build the workflow like

```go
for {
  // ...
  if messageCount > 100 {
    return workflow.ContinueAsNew(...)
  }
}
```

Any suggest on history size practice and when to ContinueAsNew ?

**Comment by longquanzheng** [comment, marked as answer]
Cadence has warning at 50k events and error at 200k(which would kill the workflow). Though they can be changed by config. 

I would say its recommended to continue as new at a few thousands signals if possible

**Reply by wenerme** [reply]
Thanks, how about add a faq to doc about these limits and suggests ?
